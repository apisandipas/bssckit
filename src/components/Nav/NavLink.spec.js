import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';

import NavLink from './NavLink';

describe('NavLink compomponent', () => {
  it('matches Navlink', () => {
    const { container } = render(<NavLink />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches NavLink tabs active', () => {
    const { container } = render(<NavLink tabs active />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches NavLink tabs disabled', () => {
    const { container } = render(<NavLink tabs disabled />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches NavLink tabs', () => {
    const { container } = render(<NavLink tabs />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches NavLink disabled', () => {
    const { container } = render(<NavLink disabled />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches NavLink noRadius', () => {
    const { container } = render(<NavLink to='#' noRadius />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('border-radius', '0');
  });

  it('matches NavLink pills active', () => {
    const { container } = render(<NavLink to='#' pills active />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('border-radius', '0.25rem');
    expect(container.firstChild).toHaveStyleRule('color', '#fff');
    expect(container.firstChild).toHaveStyleRule('background-color', '#007bff');
    expect(container.firstChild).toHaveStyleRule('color', '#fff', {
      modfier: '&:hover',
    });
    expect(container.firstChild).toHaveStyleRule('color', '#fff', {
      modfier: '&:focus',
    });
  });
});
