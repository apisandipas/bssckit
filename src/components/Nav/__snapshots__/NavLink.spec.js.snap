// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink compomponent matches NavLink disabled 1`] = `
.c0 {
  display: block;
  padding: 0.5rem;
  border: 1px solid transparent;
  color: #abaeb1;
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:focus {
  color: #abaeb1;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<a
  class="sc-AxirZ c0"
  disabled=""
/>
`;

exports[`NavLink compomponent matches NavLink noRadius 1`] = `
.c0 {
  display: block;
  padding: 0.5rem;
  border: 1px solid transparent;
  color: #007bff;
  border-radius: 0;
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:focus {
  color: #0069d9;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<a
  class="sc-AxirZ c0"
  to="#"
/>
`;

exports[`NavLink compomponent matches NavLink pills active 1`] = `
.c0 {
  display: block;
  padding: 0.5rem;
  border: 1px solid transparent;
  color: #fff;
  background-color: #007bff;
  background-color: #007bff;
  border-radius: 0.25rem;
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:focus {
  color: #fff;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<a
  class="sc-AxirZ c0"
  to="#"
/>
`;

exports[`NavLink compomponent matches NavLink tabs 1`] = `
.c0 {
  display: block;
  padding: 0.5rem;
  border: 1px solid transparent;
  color: #007bff;
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:focus {
  color: #0069d9;
}

.c0:hover,
.c0:focus {
  border-color: #d6d8d9 #d6d8d9 #d6d8d9;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<a
  class="sc-AxirZ c0"
/>
`;

exports[`NavLink compomponent matches NavLink tabs active 1`] = `
.c0 {
  display: block;
  padding: 0.5rem;
  border: 1px solid transparent;
  color: #343a40;
  border-color: #d6d8d9 #d6d8d9 #fff;
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:focus {
  color: #201a1d;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<a
  class="sc-AxirZ c0"
/>
`;

exports[`NavLink compomponent matches NavLink tabs disabled 1`] = `
.c0 {
  display: block;
  padding: 0.5rem;
  border: 1px solid transparent;
  color: #abaeb1;
  border-color: transparent;
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:focus {
  color: #abaeb1;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<a
  class="sc-AxirZ c0"
  disabled=""
/>
`;

exports[`NavLink compomponent matches Navlink 1`] = `
.c0 {
  display: block;
  padding: 0.5rem;
  border: 1px solid transparent;
  color: #007bff;
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.c0:hover,
.c0:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover,
.c0:focus {
  color: #0069d9;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<a
  class="sc-AxirZ c0"
/>
`;
